trace/doc.go

```diff
@@ -0,0 +1,164 @@
+// Package trace provies an API for initiating and reporting SSF
+// traces and attaching spans to them. Veneur's tracing API also
+// provides an experimental opentracing compatibility layer.
+//
+// The Veneur tracing API is completely independent of the opentracing
+// compatibility layer, with the exception of the function
+// StartSpanFromContext.
+//
+// Setup and Structure of the API
+//
+// To use the API, user code must set its service name via the global
+// variable Service, which is used by trace visualization tools to
+// tell apart the spans reported by different services. Code using the
+// trace API must set its Service, and set it only once, in main():
```

Comment:
I don't think we recommend setting this in the main function; in fact, I think we recommend setting this at the top-level or in `init()`, so it's already accessible in `main()`.

We *do* recommend only setting it in the main *package*, which is probably a distinction worth calling out here - libraries can (and should) incorporate the tracing API, but they should *not* set the service.

Reply:
Good point. Willdo.